agent/artifact_uploader.go

```diff
@@ -47,6 +47,9 @@ type ArtifactUploaderConfig struct {
 
 	// Whether to follow symbolic links when resolving globs
 	FollowSymlinks bool
+
+	// Whether to not upload symlinks
+	NoUploadSymlinks bool
```

Comment:
We need to invert the field as existing behaviour is equivalent to setting `NoUploadSymlinks` to false, and when creating a new `ArtifactUploaderConfig` struct, the field will be `false` by default.